cloudinit/distros/__init__.py

```diff
@@ -274,6 +275,66 @@ def _write_network(self, settings):
             "_write_network_config needs implementation.\n" % self.name
         )
 
+    @property
+    def dhcp_client(self) -> dhcp.DhcpClient:
+        """access the distro's preferred dhcp client
+
+        if no client has been selected yet select one - uses
+        self.dhcp_client_priority, which may be overriden in each distro's
+        object to eliminate checking for clients which will not be provided
+        by the distro
+        """
+        if self._dhcp_client:
+            return self._dhcp_client
+
+        # no client has been selected yet, so pick one
+        #
+        # set the default priority list to the distro-defined priority list
+        dhcp_client_priority = self.dhcp_client_priority
+
+        # if the configuration includes a network.dhcp_client_priority list
+        # then attempt to use it
+        config_priority = util.get_cfg_by_path(
+            self._cfg, ("network", "dhcp_client_priority"), []
+        )
+
+        if config_priority:
+            # user or image builder configured a custom dhcp client priority
+            # list
+            found_clients = []
+            LOG.debug(
+                "Using configured dhcp client priority list: %s",
+                config_priority,
+            )
+            for client_configured in config_priority:
+                for client_class in dhcp.ALL_DHCP_CLIENTS:
```

Comment:
This should be the distro-supported list of clients right in case subclasses reduce support from ALL_DHCP_CLIENTS.
```suggestion
                for client_class in self.dhcp_client_priority:
```

Reply:
I could go either way on that one. The distro priority list might set a single expected dhcp client, but if the user gets the binary installed and wants to use it, I think that I'd rather just let them use it if they set that in the configuration file.

Reply:
+1 I'm good with that justification.